verifyFiles.py

```diff
@@ -0,0 +1,193 @@
+	reference_set=set(reference_dict[keys_list[0]].keys())
+	for item in study_folder_details_dict_list:
+              dictionary=item[keys_list[index]]
+	      keys_from_each_dictionary=set(dictionary.keys())
+	      #if common set is empty
+	      if not common_set:
+		 #set intersection(&) to find out the common values between a set of keys
+	         common_set=reference_set & keys_from_each_dictionary
+	      else:
+		 common_set=common_set & keys_from_each_dictionary
+	      index+=1
+	#The union of all the files present in all the folders under different conditions
+	common_files_list=list(common_set)
+    	return common_files_list
+
+#Method generate_missing_files will create a list containing the files 
+#that are not common to all the subject folders processed under various conditions. 
+#
+#Input parameters: study_folder_details_dict_list,file_with_directory_details and common_files_list
+def generate_missing_files_list(study_folder_details_dict_list,file_with_directory_details,common_files_list):
+	missing_files_list=[]
+	keys_list=read_contents_from_file(file_with_directory_details)
+	index=0;
+	keys_from_all_files=set()
+	for item in study_folder_details_dict_list:
+	     dictionary=item[keys_list[index]]
+	     keys_from_individual_files=set(dictionary.keys())
+             #set union(|) to join all the values between a set of keys 
+	     keys_from_all_files=keys_from_all_files | keys_from_individual_files
+	     index+=1
+	#Missing files list is the files remaining when the common list of files is removed from the list of all the
+	#files processed under different conditions.
+	missing_files_list=list(keys_from_all_files - set(common_files_list))
+	return missing_files_list
+	     
+
+def main():
+        parser=argparse.ArgumentParser(description='verifyFiles.py', usage='./verifyFiles.py <input_file_name>',formatter_class=argparse.RawTextHelpFormatter)
+        parser.add_argument('file_in', help= textwrap.dedent('''Input the text file containing the path to the subject folders
+                                             Each directory contains subject folders which contains the unprocessed data which will be processed under  different conditions
+					     Conditions refer to the operating system  on which the process is ran or the version of the pipeline which is used to process the data.
+                                             An example would be a directory containing the files processed using CentOS6 operating system and PreFreeSurfer version 5.0.6
+                                             Sample of the input file
+                                             /home/$(USER)/CentOS6.FSL5.0.6
+                                             /home/$(USER)/CentOS7.FSL5.0.6
+                                             Each directory will contain subject folders like 100307,100308 etc'''))
+        args=parser.parse_args()
+        study_folder_details_dict_list=populate_study_folder_dict(sys.argv[1])
```

Comment:
you should assign sys.argv[1] to a variable and use it subsequently. Also, argparse allows to get the inputs by id, e.g. 'file_in' instead of position.